tests/integration/controllers/comments/deleteComment.test.js

```diff
@@ -0,0 +1,121 @@
+import { createComment } from "~/controllers/comments/create";
+import { deleteComment } from "~/controllers/comments/delete";
+import { randomAccessToken } from "./../../../utils";
+
+import { db } from "~/utils/db.server";
+
+describe('delete comment controller', () => {
+    const dbDeleteManyCommentSpy = jest.spyOn(db.Comments, "deleteMany");
+
+    it('returns an error when accessToken is not provided', async () => {
+        const deleteCommentBody = {
+            commentId: 1001,
+            accessToken: null,
+            userEmail: "miguel.cardona@wizeline.com",
+        };
+
+        const deleteCommentResponse = await deleteComment(deleteCommentBody);
+        expect(deleteCommentResponse).toBeDefined();
+        expect(deleteCommentResponse.errors).toBeDefined();
+        expect(deleteCommentResponse.errors[0].message).toBe('The provided parameters for the operation are not valid');
+        expect(deleteCommentResponse.errors[0].detail.toString()).toContain('accessToken');
+        expect(dbDeleteManyCommentSpy).toHaveBeenCalledTimes(0);
+    });
+
+    it('returns an error when provided user email does not match with the comment author', async () => {
+        const deleteCommentBody = {
+          commentId: 11,
+          accessToken: randomAccessToken(),
+          userEmail: "not.matching@wizeline.com",
+        };
+
+        const deleteCommentResponse = await deleteComment(deleteCommentBody);
+        expect(deleteCommentResponse).toBeDefined();
+        expect(deleteCommentResponse.error).toBeDefined();
+        expect(deleteCommentResponse.error.message).toBe('Error trying to delete the comment');
+        expect(deleteCommentResponse.error.detail).toBe('Comment not found or user does not have deletion rights over the comment');
+        expect(dbDeleteManyCommentSpy).toHaveBeenCalledTimes(1);
+    });
+
+    it('returns an error when session hash matches but createdAt is out of range', async () => {
+        const deleteCommentBody = {
+          commentId: 12,
+          accessToken: '8b1a5ca556d49fc46c169b2c94a058065d566f01bb208519cffd2d7419f4ff8cb4d63a060019e86354708c55766a60a2449edff151bb34681d4012699ebc3b22',
```

Comment:
How did you get this accessToken out of curiosity? We could investigate how to mock the generateSessionHash method to make it easier for tests in the future

Reply:
I just used a value coming from the *randomAccessToken* function in utils, I needed a static value for the session hash to match with the one provided to the comment in the respective fixture file or with the one generated in the test comment created using the same static accessToken